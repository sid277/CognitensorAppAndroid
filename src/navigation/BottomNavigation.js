import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {
  DashboardStackNavigator,
  AccountStackNavigator,
  NotificationStackNavigator,
} from './StackNavigation';

const Tab = createBottomTabNavigator();

const MainBottomNavigator = () => {
  const HomeScreenName = 'Home';
  const AccountScreenName = 'Account';
  const NotificationScreenName = 'Notifications';

  return (
    <Tab.Navigator
      screenOptions={({ route }) => ({
        tabBarIcon: ({ focused, color, size }) => {
          let iconName;
          if (route.name === HomeScreenName) {
            iconName = 'home';
          } else if (route.name === AccountScreenName) {
            iconName = 'account-circle';
          } else if (route.name === NotificationScreenName) {
            iconName = 'bell';
          }
          return (
            <MaterialCommunityIcons name={iconName} size={size} color={color} />
          );
        },
        headerShown: true,
      })}>
      <Tab.Screen name={HomeScreenName} component={DashboardStackNavigator} />
      <Tab.Screen name={AccountScreenName} component={AccountStackNavigator} />
      <Tab.Screen
        name={NotificationScreenName}
        component={NotificationStackNavigator}
      />
    </Tab.Navigator>
  );
};

export { MainBottomNavigator };
