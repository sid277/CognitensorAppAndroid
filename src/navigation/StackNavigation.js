import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import Dashboards from '../screens/Dashboards';
import Account from '../screens/account/Account';
import DashboardActivities from '../screens/account/DashboardActivities';
import Feedback from '../screens/account/Feedback';
import Support from '../screens/account/Support';
import Settings from '../screens/account/Settings';
import Notification from '../screens/Notification';
import AppHeader from '../components/AppHeader';

const Stack = createStackNavigator();

const DashboardStackNavigator = () => {
  return (
    <Stack.Navigator
      headerMode="screen"
      screenOptions={{
        header: ({ scene, previous, navigation }) => (
          <AppHeader
            scene={scene}
            previous={previous}
            navigation={navigation}
            searchVisible={true}
          />
        ),
      }}>
      <Stack.Screen name="Home" component={Dashboards} />
    </Stack.Navigator>
  );
};

const AccountStackNavigator = () => {
  return (
    <Stack.Navigator
      headerMode="screen"
      screenOptions={{
        header: ({ scene, previous, navigation }) => (
          <AppHeader
            scene={scene}
            previous={previous}
            navigation={navigation}
          />
        ),
      }}>
      <Stack.Screen name="Account" component={Account} />
      <Stack.Screen
        name="Dashboard Activities"
        component={DashboardActivities}
      />
      <Stack.Screen name="Feedback" component={Feedback} />
      <Stack.Screen name="Support" component={Support} />
      <Stack.Screen name="Settings" component={Settings} />
    </Stack.Navigator>
  );
};

const NotificationStackNavigator = () => {
  return (
    <Stack.Navigator
      headerMode="screen"
      screenOptions={{
        header: ({ scene, previous, navigation }) => (
          <AppHeader
            scene={scene}
            previous={previous}
            navigation={navigation}
          />
        ),
      }}>
      <Stack.Screen name="Notifications" component={Notification} />
    </Stack.Navigator>
  );
};

export {
  DashboardStackNavigator,
  AccountStackNavigator,
  NotificationStackNavigator,
};
