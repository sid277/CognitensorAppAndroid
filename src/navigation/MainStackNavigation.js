import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { MainBottomNavigator } from '../navigation/BottomNavigation';
import Login from '../screens/Login';
import Splash from '../screens/Splash';

const Stack = createStackNavigator();

const MainStackNavigator = ({ authState }) => {
  return (
    <Stack.Navigator
      screenOptions={{
        headerShown: false,
      }}>
      {authState.isLoading ? (
        <Stack.Screen name="Splash" component={Splash} />
      ) : authState.userToken == null ? (
        <Stack.Screen name="Login" component={Login} />
      ) : (
        <Stack.Screen name="Home" component={MainBottomNavigator} />
      )}
    </Stack.Navigator>
  );
};

export { MainStackNavigator };
