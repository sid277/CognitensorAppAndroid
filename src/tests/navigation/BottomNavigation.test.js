import React from 'react';
import { shallow } from 'enzyme';
import {MainBottomNavigator} from '../../navigation/BottomNavigation';
import renderer from 'react-test-renderer';


test('should test MainBottomNavigator component', () => {
const wrapper = MainBottomNavigator();
expect(wrapper).toMatchSnapshot();
});