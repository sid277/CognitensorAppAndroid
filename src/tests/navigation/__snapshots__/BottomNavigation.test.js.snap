// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test MainBottomNavigator component 1`] = `
<BottomTabNavigator
  screenOptions={[Function]}
>
  <Screen
    component={[Function]}
    name="Home"
  />
  <Screen
    component={[Function]}
    name="Account"
  />
  <Screen
    component={[Function]}
    name="Notifications"
  />
</BottomTabNavigator>
`;
